lib/libc/common/Kconfig

```
@@ -6,11 +6,44 @@ config COMMON_LIBC_ABORT
 	help
 	  common implementation of abort().
 
+config COMMON_LIBC_ASCTIME
+	bool
+	help
+	  common implementation of asctime().
+
+config COMMON_LIBC_ASCTIME_R
+	bool "Thread-safe version of asctime()"
+	default y if POSIX_THREAD_SAFE_FUNCTIONS
+	select COMMON_LIBC_ASCTIME
+	help
+	  common implementation of asctime_r().
+
+config COMMON_LIBC_CTIME
+	bool
+	select COMMON_LIBC_LOCALTIME_R_UTC
+	help
+	  common implementation of ctime().
+
+config COMMON_LIBC_CTIME_R
+	bool "Thread-safe version of ctime()"
+	default y if POSIX_THREAD_SAFE_FUNCTIONS
+	select COMMON_LIBC_CTIME
+	help
+	  common implementation of ctime_r().
+
 config COMMON_LIBC_GMTIME_R
 	bool
 	help
 	  common implementation of gmtime_r().
 
+config COMMON_LIBC_LOCALTIME_R_UTC
```

I think this one can be abbreviated without the UTC. 

From [here](https://pubs.opengroup.org/onlinepubs/9799919799/functions/localtime.html)
> Unlike localtime(), the localtime_r() function is not required to set tzname. If localtime_r() sets tzname, it shall also set daylight and timezone. If localtime_r() does not set tzname, it shall not set daylight and shall not set timezone. If the tm structure member tm_zone is accessed after the value of TZ is subsequently modified, the behaviour is undefined.

So if our implementation of `localtime_r()` never sets the timezone (i.e. is UTC-only), it's still compliant. No special treatment necessary. My assumption is that most would probably like to avoid bringing timezone details into Zephyr.

```suggestion
config COMMON_LIBC_LOCALTIME_R
```

---

`COMMON_LIBC_LOCALTIME_R_UTC` is shared by both the `localtime()` & `localtime_r()`, I guess that's why I added that suffix to make it obvious that both functions only work for UTC, should I still remove the `_UTC` suffix? Or maybe I should create another Kconfig for `localtime()`?